datasetinsights/estimators/faster_rcnn.py

```diff
@@ -355,619 +884,123 @@ def _gt_preds2tensor(gt_preds, max_boxes=MAX_BOXES_PER_IMAGE):
-            )
-        self.model_without_ddp.eval()
 
-    def predict(self, pil_img, box_score_thresh=0.5):
-        """ Get prediction from one image using loaded model
+class ToTensor:
+    def __call__(self, image, target):
+        image = torchvision.transforms.functional.to_tensor(image)
+        return image, target
 
-        Args:
-            pil_img (PIL Image): PIL image from dataset.
-            box_score_thresh (float): box score threshold for filter out lower
-            score bounding boxes. Defaults to 0.5.
 
-        Returns:
-            filtered_pred_annotations (List[BBox2D]): high predicted score
-            bboxes from the model.
-        """
-        pil_img = pil_img.unsqueeze(0)
-        img_tensor = pil_img.to(self.device)
-        predicts = self.model_without_ddp(img_tensor)
-        predict_annotations = convert_bboxes2canonical(predicts)[0]
-        filtered_pred_annotations = [
-            box for box in predict_annotations if box.score >= box_score_thresh
-        ]
-        return filtered_pred_annotations
+def get_transform():
+    transforms = [PrepBoxes(), ToTensor()]
+    return Compose(transforms)
 
-    def train_loop(
-        self,
-        *,
-        model,
-        config,
-        writer,
-        kfp_writer,
-        train_dataloader,
-        label_mappings,
-        val_dataloader,
-        train_sampler=None,
-    ):
-        """
 
-        Args:
-            model: model to train
-            config (CfgNode): estimator config
-            writer: Tensorboard writer object
-            device: model training on device (cpu|cuda)
-            train_dataloader (torch.utils.data.DataLoader):
-            label_mappings (dict): a dict of {label_id: label_name} mapping
-            val_dataloader (torch.utils.data.DataLoader):
-        """
-        model = model.to(self.device)
-        params = [p for p in model.parameters() if p.requires_grad]
-        if config.optimizer.name == "Adam":
-            optimizer = torch.optim.Adam(params, **config.optimizer.args)
-
-            # use fixed learning rate when using Adam
-            lr_scheduler = torch.optim.lr_scheduler.LambdaLR(
-                optimizer, lambda x: 1.0
-            )
-        elif config.optimizer.name == "SGD":
-            optimizer = torch.optim.SGD(
-                params,
-                lr=config.optimizer.args.lr,
-                momentum=config.optimizer.args.momentum,
-                weight_decay=float(config.optimizer.args.weight_decay),
-            )
-            lr_scheduler = torch.optim.lr_scheduler.StepLR(
-                optimizer,
-                step_size=config.optimizer.args.lr_step_size,
-                gamma=config.optimizer.args.lr_gamma,
-            )
-        else:
-            raise ValueError(
-                f"only valid optimizers are 'SGD' and 'Adam' but "
-                f"received paramater {config.optimizer.name}"
-            )
+def collate_fn(batch):
```

Comment:
Is this method required as it's only a 1-line method?

Reply:
yeah its one line, it was from beginning, Let me see If I can remove.

Reply:
So should this method kept in faster_rcnn module? Maybe it does not belong here?